### PR TITLE
HIP-30 Boilerplate

### PR DESCRIPTION
**Pull request by MaxMustermann2**
This pull request contains
- Addition of HIP-30 parameters to the sharding configuration: emission split and emission destination address
- A hard fork scheduled sometime TBD in the future for HIP-30, at which point `genesis.HarmonyAccountsPostHIP30` goes into effect. It contains our existing keys except those which belonged to shards 2 and 3 per the original formula.
- Minor comment edits in `core/state/statedb.go`
- Largely cosmetic changes to the many `MustNewInstance` lines. Earlier, they were extending across the screen's viewport but now they have been broken down into multiple lines for improved readability.

### CODE REVIEW

**Review by ONECasey — APPROVED** [review]

**Review by ONECasey — APPROVED** [review]

File: internal/configs/sharding/instance.go [hunk 1]
```diff
@@ -37,18 +37,26 @@ type instance struct {
...
+	numNodesPerShard,
+	numHarmonyOperatedNodesPerShard,
+	slotsLimit int,
+	harmonyVotePercent numeric.Dec,
 	hmyAccounts []genesis.DeployAccount,
 	fnAccounts []genesis.DeployAccount,
 	allowlist Allowlist,
 	feeCollectors FeeCollectors,
+	emissionFractionToRecovery numeric.Dec,
+	recoveryAddress ethCommon.Address,
```
Comment (sophoah) [inline internal/configs/sharding/instance.go:59]: why is recovery address not a configuration as part of sharding config like here https://github.com/harmony-one/harmony/blob/1b9614ba240de1fb431c8f800bb378f2358ae7eb/internal/configs/sharding/mainnet.go#L52

Comment (sophoah) [reply internal/configs/sharding/instance.go:59]: let's leave this here as I am thinking  we may in the future change to another address eventually.

**Review by sophoah — APPROVED** [review]
LGTM

**Comment by sophoah** [comment]
note something important was the HarmonyAccountsPostHIP30 which I've confirmed is correct.

**Comment by sophoah** [comment]
@ONECasey if we do two hardfork instead of 1, this one will definitely require some updates, however we need to see those two hardforks globally with all the PRs submitted by Max

**Comment by ONECasey** [comment]
> @ONECasey if we do two hardfork instead of 1, this one will definitely require some updates, however we need to see those two hardforks globally with all the PRs submitted by Max

I say we try and put all HIP-30 updates into 1 fork

**Pull request merged**